src/main/java/com/ibm/eventstreams/connect/mqsource/builders/BaseRecordBuilder.java

```diff
@@ -1,199 +1,174 @@
-
-        return sourceRecord;
-    }
-
-    public ConnectHeaders convertJmsPropertiesToKafkaHeaders(boolean messageBodyJms, Message message){
-
-        ConnectHeaders connectHeaders = new ConnectHeaders();
-
-        if (messageBodyJms==true) {
-
-            try {
-                Enumeration jmsPropertyNames = message.getPropertyNames();
-                while (jmsPropertyNames.hasMoreElements()) {
-                    Object jmsPropertyKey = jmsPropertyNames.nextElement();
-                    String jmsPropertyValue = message.getStringProperty(jmsPropertyKey.toString());
-                    connectHeaders.addString(jmsPropertyKey.toString(), jmsPropertyValue);
-
-                }
-            }
-            catch (JMSException e){
-                log.error("MQ/JMS message header could not be read", e);
-            }
-
-
-        }
-
-        return connectHeaders;
-
-    }
+	private static final Logger log = LoggerFactory.getLogger(BaseRecordBuilder.class);
+	protected KeyHeader keyheader = KeyHeader.NONE;
+	private Boolean copyJmsPropertiesFlag = MQSourceConnector.CONFIG_VALUE_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER_DEFAULT;
+	private JmsToKafkaHeaderConverter jmsToKafkaHeaderConverter;
+
+	/**
+	 * Configure this class.
+	 *
+	 * @param props initial configuration
+	 *
+	 * @throws ConnectException   Operation failed and connector should stop.
+	 */
+	@Override
+	public void configure(Map<String, String> props) {
+		log.trace("[{}] Entry {}.configure, props={}", Thread.currentThread().getId(), this.getClass().getName(), props);
+
+		String kh = props.get(MQSourceConnector.CONFIG_NAME_MQ_RECORD_BUILDER_KEY_HEADER);
+		if (kh != null) {
+			if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_JMSMESSAGEID)) {
+				keyheader = KeyHeader.MESSAGE_ID;
+				log.debug("Setting Kafka record key from JMSMessageID header field");
+			} else if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_JMSCORRELATIONID)) {
+				keyheader = KeyHeader.CORRELATION_ID;
+				log.debug("Setting Kafka record key from JMSCorrelationID header field");
+			} else if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_JMSCORRELATIONIDASBYTES)) {
+				keyheader = KeyHeader.CORRELATION_ID_AS_BYTES;
+				log.debug("Setting Kafka record key from JMSCorrelationIDAsBytes header field");
+			} else if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_JMSDESTINATION)) {
+				keyheader = KeyHeader.DESTINATION;
+				log.debug("Setting Kafka record key from JMSDestination header field");
+			} else {
+				log.error("Unsupported MQ record builder key header value {}", kh);
+				throw new ConnectException("Unsupported MQ record builder key header value");
+			}
+		}
+
+		String str = props.get(MQSourceConnector.CONFIG_NAME_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER);
+		if (str != null && ("true".equalsIgnoreCase(str) || "false".equalsIgnoreCase(str))) {
```

Comment:
Use Boolean.parseBoolean(String) instead of literal "true" and "false".

Reply:
@AndrewJSchofield Changed as per the suggestion.